returnn/hdf.py

```diff
@@ -346,6 +360,288 @@ def run(self):
+                    audio_file,
+                    segments,
+                    compress_format=compress_format,
+                    compress_factor=compress_factor,
+                    multi_channel_strategy=multi_channel_strategy,
+                    output_dtype=output_dtype,
+                    rounding_strategy=rounding_strategy,
+                    round_factor=round_factor,
+                    target_sampling_rate=target_sampling_rate,
+                )
+            except Exception as exc:
+                if i < max_tries - 1:
+                    _logging.warning(f"Failed to process audio {audio_file}: {exc}")
+                else:
+                    _logging.warning(f"Failed to process audio {audio_file} after {max_tries} attempts. Skipping.")
+        return []
+
+    @staticmethod
+    def __process_seq(
+        audio_file: str,
+        segments: Sequence[Tuple[str, float, float]],
+        *,
+        compress_format: Optional[str],
+        compress_factor: Optional[float],
+        multi_channel_strategy: BlissToPcmHDFJob.BaseStrategy,
+        output_dtype: str,
+        rounding_strategy: BlissToPcmHDFJob.RoundingScheme,
+        round_factor: int,
+        target_sampling_rate: int,
+    ) -> List[Tuple[str, np.ndarray]]:
+        _logging.info(f"Processing {audio_file} with {len(segments)} segments.")
+
+        assert isinstance(multi_channel_strategy, (BlissToPcmHDFJob.Mixdown, BlissToPcmHDFJob.PickNth)), (
+            "unknown multi_channel_strategy"
+        )
+        channel_mix_args = (
+            ["-ac", "1"]
+            if isinstance(multi_channel_strategy, BlissToPcmHDFJob.Mixdown)
+            else ["-map_channel", f"0.{multi_channel_strategy.channel}"]
+        )
+        results = []
+
+        with tempfile.TemporaryDirectory() as tmp_dir:
+            # We preprocess all data with ffmpeg because it is more robust to different
+            # audio formats like gsm-ms (which soundfile cannot read) or mp3 (standard format,
+            # but soundfile cannot seek). We also resample it at the same time and merge channels.
+            #
+            # We still use soundfile to extract the segments because we want to slice the audio
+            # data on a frame-by-frame basis (to match with RASR), while ffmpeg supports only
+            # temporal slices.
+            audio_file_out = os.path.join(tmp_dir, os.path.basename(audio_file) + ".wav")
+            sp.check_call(
+                [
+                    "ffmpeg",
+                    "-y",
+                    "-i",
+                    audio_file,
+                ]
+                + channel_mix_args
+                + [
+                    "-af",
+                    "aresample=resampler=soxr",
+                    "-ar",
+                    str(target_sampling_rate),
+                    audio_file_out,
+                ],
+                stdout=sp.DEVNULL,
+                stderr=sp.DEVNULL,
```

Comment:
Why? Wouldn't it be good to see errors? I guess you don't want to spam your log? Maybe just add `-hide_banner -loglevel error` to the args instead?